ibis-vzla/puppeteer-utilz
# v2 - code refactor + unit tests

## internals

### find-frame

- code refactor ✔️
- tests
  - should look for a frame within an array ✔️
  - should throw undefined if the frame is not found ✔️

### logger

- code refactor ✔️
- tests
  - should send log messages ✔️

### time

- code refactor ✔️
- tests
  - should convert string to number if a string is received ✔️
  - should do nothing if a number is received ✔️

## utils

### async-foreach

- code refactor ✔️
- tests
  - should be possible to execute async functions ✔️

### block-resources

- code refactor ✔️
- tests ❌

### capture

- code refactor ✔️
- tests
  - should take a screenshot ✔️
  - should return null if selector doesn't match any element in the DOM ✔️
  - should save screenshot in the correct path ✔️

### click

- code refactor ✔️
- tests
  - should click on a selector ✔️
  - should throw false if selector not found ✔️

### delay

- code refactor ✔️
- tests
  - with number type parameter should only return number ✔️
  - with string type parameter should only return number ✔️

### disable-logger

- code refactor ✔️
- tests
  - should disable the logger ✔️

### find-element

- code refactor ✔️
- tests
  - should find an element by content ✔️
  - should throw null if element not found ✔️

### get-frame

- code refactor ✔️
- tests
  - should return a frame ✔️
  - should return null if frame not found ✔️

### launch-browser

- code refactor ✔️
- tests
  - should set an environment ✔️

### noop

- code refactor ✔️
- tests
  - call noop should do nothing ✔️
  - call noop shouldn't do anything other except nothing ✔️

### retry

- code refactor ✔️
- tests ✔️

### set-download-path

- code refactor ✔️
- tests
  - should set a download path ✔️

### timeout

- code refactor ✔️
- tests
  - with number type parameter should only return an object with `timeout: number` ✔️
  - with string type parameter should only return an object with `timeout: number` ✔️

### type

- code refactor ✔️
- tests
  - should type into input ✔️
  - should throw Error if selector not found ✔️

### wait-for-download

- code refactor ✔️
- tests
  - should wait for a download to complete ✔️

### wait-for-frame

- code refactor ✔️
- tests
  - should wait for a frame ✔️

### wait-for-navigation

- code refactor ✔️
- tests
  - should wait for a loaded web ✔️

### wait-for-url

- code refactor ✔️
- tests
  - should wait for a redirect ✔️
